script.js

```diff
@@ -1,117 +1,133 @@
+function calculateFieldLines() {
+  let paths = [];
+  for (let i = 0; i < charges.length; i++) {
+    let startAngle = TAU * Math.random();
+    let angleStep = TAU / Math.floor(1 + Math.abs(charges[i][2]));
+    for (let j = 0; j <= Math.abs(charges[i][2]); j++)
+      paths.push(calculatePath([charges[i][0] + Math.sqrt(Math.abs(charges[i][2])) * Math.cos(startAngle + j * angleStep), charges[i][1] + Math.sqrt(Math.abs(charges[i][2])) * Math.sin(startAngle + j * angleStep), charges[i][2]]));
   }
+  return paths;
 }
 
 function calculatePath(startPoint) {
-  if (startPoint[1] > window.innerHeight || startPoint[1] < 0 || startPoint[0] > window.innerWidth || startPoint[0] < 0) {
+  if (startPoint[1] > window.innerHeight || startPoint[1] < 0 || startPoint[0] > window.innerWidth || startPoint[0] < 0)
     return [startPoint];
-  }
   let fieldStrength = [0, 0];
   for (var i = 0; i < charges.length; i++) {
-    let distanceSquare = (startPoint[0] - charges[i][0]) ** 2 + (startPoint[1] - charges[i][1]) ** 2;
-    if (distanceSquare < 25) {
+    let distanceSquare = (startPoint[0] - charges[i][0])**2 + (startPoint[1] - charges[i][1])**2;
+    if (distanceSquare < Math.abs(charges[i][2]) - 1)
       return [startPoint];
-    }
-    let magnitude = Math.sign(startPoint[0] - charges[i][0]) * charges[i][2] * Math.sign(startPoint[2]) / distanceSquare;
-    fieldStrength[0] += magnitude * Math.cos(Math.atan((startPoint[1] - charges[i][1]) / (startPoint[0] - charges[i][0])));
-    fieldStrength[1] += magnitude * Math.sin(Math.atan((startPoint[1] - charges[i][1]) / (startPoint[0] - charges[i][0])));
+    addStrength(startPoint, fieldStrength, charges[i], distanceSquare);
   }
   fieldStrength = [fieldStrength[0] / Math.sqrt(fieldStrength[0] ** 2 + fieldStrength[1] ** 2), fieldStrength[1] / Math.sqrt(fieldStrength[0] ** 2 + fieldStrength[1] ** 2)];
   return [startPoint].concat(calculatePath([startPoint[0] + fieldStrength[0], startPoint[1] + fieldStrength[1], startPoint[2]]));
+}
+
+function addStrength(startPoint, fieldStrength, charge, distanceSquare) {
+  let magnitude = Math.sign(startPoint[0] - charge[0]) * charge[2] * Math.sign(startPoint[2]) / distanceSquare;
+  fieldStrength[0] += magnitude * Math.cos(Math.atan((startPoint[1] - charge[1]) / (startPoint[0] - charge[0])));
+  fieldStrength[1] += magnitude * Math.sin(Math.atan((startPoint[1] - charge[1]) / (startPoint[0] - charge[0])));
+}
+
+function drawPath(paths, i) {
+  let ctx = document.getElementById('lines').getContext('2d');
+  ctx.beginPath();
+  if (Math.sign(paths[i][0][2]) === 1)
+    ctx.strokeStyle = "#900000";
+  else
+    ctx.strokeStyle = "#000090";
+  ctx.moveTo(paths[i][0][0], paths[i][0][1]);
+  for (var j = 0; j < paths[i].length; j++) {
+    ctx.lineTo(paths[i][j][0], paths[i][j][1]);
+  }
+  ctx.stroke();
+}
+
+function drawCharge(charge) {
+  let ctx = document.getElementById('lines').getContext('2d');
+  ctx.beginPath();
+  if (Math.sign(charge[2]) === 1)
```

Comment:
While it is legal to write `if` and `else` clauses with `{}` if the body is just one line, it tends to be error prone. (It's very easy to add a second line and forget to add the braces and then be mislead by the indentation, e.g.:

```
if (whatever)
  doAThing();
  addedThis();
```

Is really this:


```
if (whatever) {
  doAThing();
}
addedThis();
```
not:

```
if (whatever) {
  doAThing();
  addedThis();
}
```
When programming in languages like Javascript that allow this syntax, I typically follow the rule that I only omit-braces when I write things on one line:

```
if (whatever) doAthing();
```

Reply:
But in cases like this there's an even better solution namely the "ternary operator" which is like if `if/else` except it's an expression that evaluates to value. So you could write:

```
ctx.fillStyle = Math.sign(charge[2]) === 1 ? "#900000" : "#000090";
```

to set `ctx.fillStyle` to `"#900000"` when the condition holds and `"#000090"` when it does not.